### PR TITLE
feat(image): add logic to guess base layer for docker-cis scan

### LINKED ISSUE #3834: Add base image layer detection to compliance scanning
I raised issue #3814 due to false positives on docker-cis compliance scanning. The false positives were from a `HEALTHCHECK` layer and a bunch of `ADD` layers in the base image. The `HEALTHCHECK` false positive has been dealt with in said issue, but @knqyf263 has asked that I raise a new feature request to add the`guessBaseLayers` functionality to the docker-cis compliance scanning, to remove all the false positives of `ADD` layers in the base image.
https://github.com/aquasecurity/trivy/blob/3987a679f9520259a1b1474e81704b8aedce2ef7/pkg/fanal/artifact/image/image.go#L496

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add logic to guess base layer for docker-cis scan.
See more in #3834.

## Related issues
- Close #3834

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/artifact/image/image.go [hunk 1]
```diff
@@ -550,3 +527,31 @@ func (a Artifact) guessBaseLayers(diffIDs []string, configFile *v1.ConfigFile) [
 	}
 	return baseDiffIDs
 }
+
+func GuessBaseImageIndex(histories []v1.History) int {
```
Comment (knqyf263) [inline pkg/fanal/artifact/image/image.go:531]: Can we move it under `fanal/image/`? The existing file `image.go` or a new file, `history.go`

Comment (DmitriyLewen) [reply pkg/fanal/artifact/image/image.go]: I don't think that we need to create new file for 1 function.
I moved function to `fanal/image/image.go` in b2cce5f8feec1067de963f67de57810891d85882

File: pkg/fanal/image/image.go [hunk 2]
```diff
@@ -128,3 +129,31 @@ func LayerIDs(img v1.Image) ([]string, error) {
 	}
 	return layerIDs, nil
 }
+
+func GuessBaseImageIndex(histories []v1.History) int {
```
Comment (knqyf263) [inline pkg/fanal/image/image.go:133]: I think this comment should be moved to this function as the main logic exists here.
https://github.com/aquasecurity/trivy/blob/87844ca3506ab8907d53ce0e6cb69f9af0fdbc1b/pkg/fanal/artifact/image/image.go#L473-L501

Comment (knqyf263) [reply pkg/fanal/image/image.go:133]: Maybe keep the first line only in Artifact.guessBaseLayers().

Comment (DmitriyLewen) [reply pkg/fanal/image/image.go:162]: You are right. Made this in https://github.com/aquasecurity/trivy/pull/4344/commits/d8c4002f656d6ad32b7f2796d4e43686fc42a161

**Pull request merged**